scripts/generate_records/generate_db_records.js

```diff
@@ -0,0 +1,396 @@
+ *   filesPerGranule: number
+ *   granulesPerBatch: number,
+ *   executionsPerBatch: number,
+ *   models: ModelSet,
+ *   swallowErrors: boolean,
+ * }} BatchParams
+ *
+ * @param {BatchParams} params
+ * @returns {Promise<{
+ *   granuleCumulusIds: Array<number>
+ *   fileCumulusIds: Array<number>
+ *   executionCumulusIds: Array<number>
+ * }>}
+ */
+const uploadDataBatch = async ({
+  knex,
+  collectionCumulusId,
+  providerCumulusId,
+  filesPerGranule,
+  granulesPerBatch,
+  executionsPerBatch,
+  models,
+  swallowErrors,
+}) => {
+  const granuleCumulusIds = await loadGranules(
+    knex,
+    collectionCumulusId,
+    providerCumulusId,
+    granulesPerBatch,
+    models.granuleModel,
+    swallowErrors
+  );
+  const fileCumulusIds = [];
+  for (const granuleCumulusId of granuleCumulusIds) {
+    fileCumulusIds.push(
+      await loadFiles(knex, granuleCumulusId, filesPerGranule, models.fileModel, swallowErrors)
+    );
+  }
+  const executionCumulusIds = await loadExecutions(
+    knex,
+    collectionCumulusId,
+    executionsPerBatch,
+    models.executionModel,
+    swallowErrors
+  );
+  await loadGranulesExecutions(
+    knex,
+    granuleCumulusIds,
+    executionCumulusIds,
+    models.geModel,
+    swallowErrors
+  );
+  return {
+    granuleCumulusIds,
+    fileCumulusIds: fileCumulusIds.flat(),
+    executionCumulusIds,
+  };
+};
+
+/**
+ * create a generator Object that pretends to be an Iterable
+ * this is to allow pmap to use this data without holding the entire (potentially very large)
+ * set of batch params for more than the currently running threads
+ *
+ * @param {object} params
+ * @param {Knex} params.knex
+ * @param {number} params.numberOfGranules
+ * @param {number} params.collectionCumulusId
+ * @param {number} params.providerCumulusId
+ * @param {number} params.filesPerGranule
+ * @param {number} params.granulesPerBatch
+ * @param {number} params.executionsPerBatch
+ * @param {ModelSet} params.models
+ * @param {boolean} params.variance
+ * @param {boolean} params.swallowErrors
+ * @returns {Iterable<BatchParams>}
+ */
+
+const getBatchParamGenerator = ({
+  knex,
+  numberOfGranules,
+  collectionCumulusId,
+  providerCumulusId,
+  filesPerGranule,
+  granulesPerBatch,
+  executionsPerBatch,
+  models,
+  variance,
+  swallowErrors = false,
+}) => {
+  if (granulesPerBatch < 1) {
+    throw new Error('granulesPerBatch must be set to >=1');
+  }
+  /**
+   * @yields {BatchParams}
+   */
+  function* batchParamGenerator() {
+    const bar = new cliProgress.SingleBar(
+      { etaBuffer: numberOfGranules / 10 }
+    );
+    bar.start(numberOfGranules, 0);
+
+    const standardParams = {
+      knex,
+      collectionCumulusId,
+      providerCumulusId,
+      filesPerGranule,
+      granulesPerBatch,
+      executionsPerBatch,
+      models,
+      swallowErrors,
+    };
+
+    //asking for variance adds some noise to batch executions vs granules
+    let _granulesPerBatch = 1;
```

Comment:
I understand wanting to start this at 1, but now if there's no variance, `_granulesPerBatch` will always be 1, is that what you want?

Reply:
good catch, this was a mistake
[error fix in batch params](https://github.com/nasa/cumulus/pull/3679/commits/01273d2e1224e806d547f1c82a8013415db60fd5)